### PR TITLE
HIVE-26742: Iceberg: Vectorization fails incase of multiple tables in queries.

### PR DESCRIPTION
**Pull request by ayushtkn**
### What changes were proposed in this pull request?

Fix Vectorization for Iceberg tables

### Why are the changes needed?

Configuration getting propagated for non iceberg tables as well

### Does this PR introduce _any_ user-facing change?

Yes, The query passes

### How was this patch tested?

Added a UT

### CODE REVIEW

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java [hunk 1]
```diff
@@ -198,4 +203,10 @@ public void injectCaches(FileMetadataCache metadataCache, DataCache dataCache, C
     // no-op for Iceberg
   }
 
+  public static String getVectorizationConfName(String tableName) {
+    String[] components = tableName.split("\\.");
+    String dbAndTableName = components.length == 3 ? components[1] + "." +  components[2] : tableName;
```
Comment (szlta) [inline iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java:208]: TableName.fromString could help here?

**Review by szlta — COMMENTED** [review]
Looks good, just a nit

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java (hunk 1 above)
Comment (ayushtkn) [reply iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java]: Done

**Review by szlta — APPROVED** [review]
+1 pending tests

**Pull request merged**